src/ua_connection.c

```diff
@@ -102,6 +108,8 @@ separateIncompleteChunk(UA_Connection *connection, UA_ByteString * UA_RESTRICT m
                 return retval;
             memcpy(connection->incompleteMessage.data, message->data, message->length);
             connection->releaseRecvBuffer(connection, message);
+            //The complete message is buffered, so no more content to be processed in message
+            message->length = 0;
```

Comment:
I'm not sure if this is correct... If the message is copied into `connection->incompleteMessage` then `message` should be set to empty, because there is no more content to be processed by the following methods, especially because the content of `message` is not complete